.github/workflows/pegasus.yml

```diff
@@ -0,0 +1,96 @@
+name: GraphScope Pegasus CI
+
+on:
+  push:
+    branches:
+      - gaia-x
+    paths:
+      - 'research/engine/pegasus/**'
+      - '.github/workflows/pegasus.yml'
+
+  pull_request:
+    branches:
+      - gaia-x
+    paths:
+      - 'research/engine/pegasus/**'
+      - '.github/workflows/pegasus.yml'
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  peagsus-build:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build & Test
+      run: |
+        source ~/.bashrc
+
+        echo $(pwd)
+        cd research/engine/pegasus
+        cargo build --verbose
+        cargo test --verbose
+
+  pegasus-test:
+    runs-on: ubuntu-20.04
+    needs: [peagsus-build]
```

Comment:
Since `pegasus-test` and `pegasus-test` were dispatched to the different machines(GitHub Runner), I guess maybe you should rebuild the `peagsus`.

Reply:
`pegasus-test` and `pegasus-build` are now in a same job so that pegasus will be built only one time.